src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -210,4 +214,35 @@ void GIVEN_shadow_manager_WHEN_individual_config_resets_THEN_respond_to_config_u
 
         assertThat(syncHandler.getSyncDirection(), is(Direction.DEVICE_TO_CLOUD));
     }
+
+    @Test
+    @SuppressWarnings("PMD.CloseResource")
+    void GIVEN_shadow_manager_WHEN_strategy_config_resets_THEN_respond_to_config_updates(ExtensionContext context)
+            throws Exception {
+        ignoreExceptionOfType(context, SkipSyncRequestException.class);
+        MqttClient mqttClient = mock(MqttClient.class);
+        lenient().when(mqttClient.connected()).thenReturn(false);
+
+
+        kernel.getContext().put(MqttClient.class, mqttClient);
+        startNucleusWithConfig(NucleusLaunchUtilsConfig.builder()
+                .configFile("periodic_sync.yaml")
+                .mqttConnected(false)
+                .syncClazz(PeriodicSyncStrategy.class)
+                .build());
+
+        SyncHandler syncHandler = kernel.getContext().get(SyncHandler.class);
+        assertThat(syncHandler.getOverallSyncStrategy(), instanceOf(PeriodicSyncStrategy.class));
```

Comment:
I know we do this in another tests but this type of testing couples tests to implementation. The tests now knows that we use a class called `SyncHandler` and it also know that the `syncHandler` has a method called `getOverallSyncStrategy` and we are asserting a specific class gets returned making it hard to change.